### PR TITLE
Completed Node Coloring MVC Move

### PR DESCRIPTION
**Pull request by mihirsamdarshi**
This PR will contain the entire node coloring codebase's move into the new architecture

### CODE REVIEW

**Comment by mihirsamdarshi** [comment]
@dondi my code is ready for review, I am finally done

File: web-client/public/js/graph.js [hunk 1]
```diff
@@ -1105,40 +1106,34 @@ export var drawGraph = function (network, nodeColoring) {
...
 
-    nodeColoring.renderNodeColoring = function () {
-        if (grnState.nodeColoring.nodeColoringEnabled) {
-            colorNodes("top", grnState.nodeColoring.topDataset, grnState.nodeColoring.avgTopDataset,
-                        grnState.nodeColoring.logFoldChangeMaxValue);
-            colorNodes("bottom", grnState.nodeColoring.bottomDataset, grnState.nodeColoring.avgBottomDataset,
-                        grnState.nodeColoring.logFoldChangeMaxValue);
-            renderNodeLabels();
-            renderNodeColoringLegend(grnState.nodeColoring.logFoldChangeMaxValue);
+    const hasExpressionData = (sheets) => {
```
Comment (dondi) [inline web-client/public/js/graph.js]: Small ES6 formatting note: when an arrow function has a single argument, the parentheses are optional.
```suggestion
    const hasExpressionData = sheets => {
```

File: web-client/public/js/graph.js [hunk 2]
```diff
@@ -1105,40 +1106,34 @@ export var drawGraph = function (network, nodeColoring) {
...
+            if (property.match(ENDS_IN_EXPRESSION_REGEXP)) {
+                return true;
+            }
         }
+        return false;
     };
 
-    if (!$.isEmptyObject(network.expression) && hasExpressionData(network.expression)) {
-        nodeColoring.renderNodeColoring();
+    if (!$.isEmptyObject(network.expression) && hasExpressionData(network.expression) &
```
Comment (dondi) [inline web-client/public/js/graph.js]:
```suggestion
    if (!$.isEmptyObject(network.expression) && hasExpressionData(network.expression) &&
```

File: web-client/public/js/update-app.js [hunk 3]
```diff
@@ -174,11 +209,229 @@ export const modifyLinkDistanceParameter = (value) => {
...
+    var createHTMLforDataset = function (name) {
+        return `
+            <li class=\"dataset-option node-coloring-menu\" value=\"${name}\">
+              <a>
+                <span class=\"glyphicon\"></span>
+                &nbsp;${name}
+              </a>
+            </li>`;
+    };
+    console.log(grnState.nodeColoring.nodeColoringOptions);
```
Comment (dondi) [inline web-client/public/js/update-app.js]: Are you keeping this console.log on purpose?

File: web-client/public/js/update-app.js [hunk 4]
```diff
@@ -174,11 +209,229 @@ export const modifyLinkDistanceParameter = (value) => {
...
+    /* eslint-enable max-len */
+    updaters.renderNodeColoring();
+};
+
+const showNodeColoringMenus = () => {
+    if ($(NODE_COLORING_MENU).hasClass("hidden")) {
+        $(NODE_COLORING_MENU).removeClass("hidden");
+    }
+    if ($(".node-coloring-menu").hasClass("disabled")) {
+        $(".node-coloring-menu").removeClass("disabled");
```
Comment (dondi) [inline web-client/public/js/update-app.js]: Is this a missed constant substitution?

**Comment by mihirsamdarshi** [comment]
@dondi the PR changes have been made

**Pull request merged**